auth/shared-signals-health-check/service/health-check-service.ts

```diff
@@ -0,0 +1,144 @@
+import type { SharedSignalsHealthCheck } from '../interface/health-check';
+import { AuthError, VerifyError } from '../errors';
+import { getSecret } from '@aws-lambda-powertools/parameters/secrets';
+import type { SecretsConfig } from '../interface/secret-config';
+
+export class SharedSignalsHealthCheckService
+  implements SharedSignalsHealthCheck
+{
+  private readonly healthCheckTokenUrl: string;
+  private readonly healthCheckVerifyUrl: string;
+  private readonly healthCheckSecretName: string;
+
+  public constructor(
+    healthCheckTokenUrl: string,
+    healthCheckVerifyUrl: string,
+    healthCheckSecretName: string,
+  ) {
+    this.healthCheckTokenUrl = healthCheckTokenUrl;
+    this.healthCheckVerifyUrl = healthCheckVerifyUrl;
+    this.healthCheckSecretName = healthCheckSecretName;
+  }
+
+  public async authorise(): Promise<string> {
+    try {
+      const secret = await getSecret(this.healthCheckSecretName);
+      if (secret === undefined) {
+        throw new AuthError(
+          `Failed to retrieve secret for ${this.healthCheckSecretName}`,
+        );
+      }
+      // prettier-ignore
+      if (typeof secret !== 'string') { //pragma: allowlist secret
+        throw new AuthError(
+          `Secret for ${this.healthCheckSecretName} is not a string`,
+        );
+      }
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-type-assertion
+      const secretConfig: SecretsConfig = JSON.parse(secret) as SecretsConfig;
+
+      const axiosConfig: AxiosRequestConfig =
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-type-assertion
+        this.constructAuthoriseAxiosRequestConfig(secretConfig);
+
+      const response = await axios<{ access_token: string }>(axiosConfig);
+
+      if (response.status !== StatusCodes.OK.valueOf()) {
+        throw new AuthError(
+          `Failed to authorise: ${response.status.toString()} ${
+            response.statusText
+          }`,
+        );
+      }
+
+      if (typeof response.data.access_token !== 'string') {
+        throw new AuthError('Access token not found in response');
+      }
+
+      return response.data.access_token;
+    } catch (error) {
+      throw new AuthError(
+        `Failed to authorise: ${
+          error instanceof Error ? error.message : 'Unknown error'
+        }`,
+      );
+    }
+  }
+
+  public async verify(token: string): Promise<boolean> {
+    try {
+      let isVerified = false;
+      const axiosConfig: AxiosRequestConfig =
+        this.constructVerifyAxiosRequestConfig(token);
+      const response = await axios(axiosConfig);
+      isVerified = response.status === StatusCodes.NO_CONTENT.valueOf();
+      if (isVerified) {
+        console.info('Token verification successful');
+      } else {
+        throw new VerifyError(
+          `Failed to verify: ${response.status.toString()} ${
+            response.statusText
+          }`,
+        );
+      }
+      return isVerified;
```

Comment:
Minor

```suggestion
      if (!isVerified) {
        throw new VerifyError(
          `Failed to verify: ${response.status.toString()} ${
            response.statusText
          }`,
        );
      }
      
      console.info('Token verification successful');
      return isVerified;
```

Unless the linter is playing up